autosklearn/ensembles/ensemble_selection.py

```diff
@@ -105,39 +105,33 @@ def _fast(
                 dtype=np.float64,
             )
             s = len(ensemble)
-            if s == 0:
-                weighted_ensemble_prediction.fill(0.0)
-            else:
-                weighted_ensemble_prediction.fill(0.0)
-                for pred in ensemble:
-                    np.add(
-                        weighted_ensemble_prediction,
-                        pred,
-                        out=weighted_ensemble_prediction,
-                    )
-                np.multiply(
+            if s > 0:
+                # Weighted ensemble prediction contains in this point the average of
+                # (s-1) elements. We do an efficient incorporation of just the new element.
+                # We incorporate the new prediction ensemble[-1] (divided by (s))
+                # We do so to just have 1 addition and 1 multiplication to integrate the new pred
+                # remember that in the past iteration we divided by s+1 all predictions in ensemble
+                # in preparation to search for a new ensemble member.
+                np.add(
                     weighted_ensemble_prediction,
-                    1/s,
+                    ensemble[-1] * (1/s),
```

Comment:
Hey, I don't think this scaling here is necessary which allows the code to be simplified even further. You can drop the scaling here, the multiply in line 120 and the multiplication in line 135. Then you'll only have to add an `np.divide(fant_ensemble_prediction, s + 1, out=fant_ensemble_prediction)` to obtain the same behavior.